### PR TITLE
Add dummy auth module

### PR DESCRIPTION
**Pull request by mkacper**
Docker container from this branch is needed.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6348.1](https://travis-ci.org/esl/MongooseIM/jobs/524037275) / Erlang 19.3 / small_tests / 32a71bf8f8634f1a90f9bdb3e84eb59ca00dbc2d
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2283/6348/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/6348/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-24_15.33.41/lib.mongooseim.logs/run.2019-04-24_15.33.42/suite.log.html)

**Comment by mongoose-im** [comment]
[6391.2](https://travis-ci.org/esl/MongooseIM/jobs/528756465) / Erlang 19.3 / internal_mnesia / 72499a7ea6bc0e75a2e940d6dd611a8991dd8f5f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2283/6391/internal_mnesia.19.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/6391/internal_mnesia.19.3/big/ct_run.test%40travis-job-6bc419ea-1d7f-4d51-b1e9-fc92858b71aa.2019-05-06_11.38.33/index.html)
**OK: 1225** / **Failed: 1** / **User-skipped: 65** / Auto-skipped: 0 
<details><summary>sm_SUITE:parallel:subscription_requests_are_buffered_properly</summary>


```erlang
{error,{{badmatch,false},
    [{escalus_session,stream_management,2,
              [{file,"/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_session.erl"},
               {line,227}]},
     {escalus_connection,connection_step,2,
               [{file,"/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_connection.erl"},
                {line,134}]},
     {lists,foldl,3,[{file,"lists.erl"},{line,1263}]},
     {escalus_connection,start,2,
               [{file,"/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_connection.erl"},
                {line,118}]},
     {sm_SUITE,'-subscription_requests_are_buffered_properly/1-fun-3-',6,
           [{file,"sm_SUITE.erl"},{line,848}]},
     {escalus_story,story,4,
            [{file,"/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
             {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
            [{file,"test_server.erl"},{line,1045}]}]}}
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/6391/internal_mnesia.19.3/big/ct_run.test%40travis-job-6bc419ea-1d7f-4d51-b1e9-fc92858b71aa.2019-05-06_11.38.33/big_tests.tests.sm_SUITE.logs/run.2019-05-06_11.52.07/sm_suite.subscription_requests_are_buffered_properly.2043.html)
</details>

**Comment by NelsonVides** [comment]
@mkacper I can see that you're making changes to this PR. I'd recommend rebasing over master, to get all the ton of changes we've recently done over CI stabilisation and CircleCI features and so on :)

**Comment by mongoose-im** [comment]
[7085.1](https://travis-ci.org/esl/MongooseIM/jobs/580141297) / Erlang 22.0 / small_tests / 218aebbe0cc798689894ceab4af38807a2dbe512
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2283/7085/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/7085/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-03_09.17.56/lib.mongooseim.logs/run.2019-09-03_09.17.57/suite.log.html)

**Comment by mkacper** [comment]
> @mkacper I can see that you're making changes to this PR. I'd recommend rebasing over master, to get all the ton of changes we've recently done over CI stabilisation and CircleCI features and so on :)

@NelsonVides Thank you for the recommendation but I would like to avoid rebasing in this specific case as I don't want to change hashes of the commits as they are referenced in other repository.

I've not introduced any big changes to the code (haven't touched `big_tests`) but if there is a problem with Travis I'd rather merge master into this branch.

**Comment by mongoose-im** [comment]
[7089.1](https://travis-ci.org/esl/MongooseIM/jobs/580192392) / Erlang 22.0 / small_tests / af7a4c7a9bb77f6aa629b7ffcb7f446a371cb498
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2283/7089/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/7089/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-03_12.22.07/lib.mongooseim.logs/run.2019-09-03_12.22.07/suite.log.html)

File: doc/authentication-backends/Dummy-authentication-module.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Overview
+
+The purpose of this module is to make it possible to authenticate a user without
+a need of real authentication. In other words, using this module allows to
```
Comment (NelsonVides) [inline doc/authentication-backends/Dummy-authentication-module.md]: Should read

>[...] a user without the need for real authentication.

Comment (mkacper) [reply doc/authentication-backends/Dummy-authentication-module.md]: Why is `the` correct here? I've search the internet for some examples and I'm not quite sure why `the` fits better than `a` here.

**Comment by mkacper** [comment]
@NelsonVides thanks for the review :) Please see the changes.

**Comment by mongoose-im** [comment]
[7090.1](https://travis-ci.org/esl/MongooseIM/jobs/580578120) / Erlang 22.0 / small_tests / e8a448579dcb179c6f66d5cd5d497c3713167899
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2283/7090/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/7090/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-04_06.56.50/lib.mongooseim.logs/run.2019-09-04_06.56.50/suite.log.html)

File: doc/authentication-backends/Dummy-authentication-module.md (hunk 1 above)
Comment (NelsonVides) [reply doc/authentication-backends/Dummy-authentication-module.md]: It was a matter of intuition, when I read the text I just felt `without the need for` would be a more correct approach. I tried looking up in the internet why do I think that is a better option but I also didn't find very clear information about the topic. Ultimately, I just asked two British native speakers and they both agreed with me, so I put it on the review 🤷‍♂

Comment (mkacper) [reply doc/authentication-backends/Dummy-authentication-module.md]: Cool, I'm convinced now 😀 Thanks!

**Comment by mongoose-im** [comment]
[7407.1](https://travis-ci.org/esl/MongooseIM/jobs/603796685) / Erlang 22.0 / small_tests / e45d3c4ab55bf77e30563f1e693a5c54603a2bc0
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2283/7407/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2283/7407/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-28_09.49.17/lib.mongooseim.logs/run.2019-10-28_09.49.18/suite.log.html)

**Pull request merged**